### ISSUE TITLE
category theory is basic

### ISSUE THREAD

**Issue by DanGrayson** [UniMath/UniMath#757]
Since category theory is so basic, I wonder whether it would be a good idea to move CategoryTheory ahead of Algebra in the list of packages, so categorical results about algebraic objects could be phrased and proved in Algebra.

The display category material could have its own package, since it's not so basic.

**Comment by benediktahrens** [comment]
On 09/20/2017 05:20 PM, Daniel R. Grayson wrote:
> Since category theory is so basic, I wonder whether it would be a
> good idea to move CategoryTheory ahead of Algebra in the list of
> packages, so categorical results about algebraic objects could be
> phrased and proved in Algebra.
> 
> The display category material could have its own package, since it's
> not so basic.

I disagree with the second proposal: one purpose of displayed categories
is to construct and reason about categories of complex structures, e.g.,
categories of algebraic structures such as groups, rings, etc.
In particular, displayed categories capture the idea of algebraic hierarchy.

Also, there is nothing non-basic about displayed categories.

In summary, I consider displayed categories
1. as part of CT.
2. as important for instances, e.g., in Algebra.

**Comment by DanGrayson** [comment]
Oh, I was assuming displayed categories were something else, thanks.  But what
is "the idea of algebraic hierarchy"?

**Comment by benediktahrens** [comment]
On 09/20/2017 05:36 PM, Daniel R. Grayson wrote:
> Oh, I was assuming displayed categories were something else, thanks.
> But what is "the idea of algebraic hierarchy"?

It refers, for instance, to the fact that rings are groups with extra
structure.
This can be mirrored by building the category of rings as the total
category of the "displayed category of rings over the base category of
groups". This displayed category contains exactly what is needed to turn
a group into a ring, and a group homomorphism into a ring homomorphism.

**Comment by rmatthes** [comment]
Are such example already implemented? Currently, I only see the explicit examples in [Examples.v](https://github.com/benediktahrens/UniMath/blob/a3912dd407adc4bcae6917283eeb3f8c94fb9780/UniMath/CategoryTheory/DisplayedCats/Examples.v)

**Comment by benediktahrens** [comment]
On 09/20/2017 06:02 PM, Ralph Matthes wrote:
> Are such example already implemented? Currently, I only see the
> explicit examples in
> [Examples.v](https://github.com/benediktahrens/UniMath/blob/a3912dd407adc4bcae6917283eeb3f8c94fb9780/UniMath/CategoryTheory/DisplayedCats/Examples.v)

No, not yet.

**Comment by benediktahrens** [comment]
Closing - any further discussion on this should be held on the mailing list.
